src/Mvc/test/WebSites/BasicWebSite/Controllers/CustomServiceApiController.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Mvc;
+
+namespace BasicWebSite;
+
+[ApiController]
+[Route("/services")]
+public class CustomServicesApiController : Controller
+{
+    [HttpGet("GetOk")]
+    public ActionResult<string> GetOk([FromKeyedServices("ok_service")] ICustomService service)
```

Comment:
For consistency, we should add a check to see if it behaves correctly when a `null` key is used.